test/performance/performance.go

```diff
@@ -0,0 +1,116 @@
+// +build performance
+
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package performance
+
+import (
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	pkgTest "github.com/knative/pkg/test"
+	"github.com/knative/serving/test"
+	"github.com/knative/test-infra/tools/testgrid"
+	"istio.io/fortio/fhttp"
+	"istio.io/fortio/periodic"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	// Mysteriously required to support GCP auth (required by k8s libs). Apparently just importing it is enough. @_@ side effects @_@. https://github.com/kubernetes/client-go/issues/242
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+)
+
+const (
+	istioNS = "istio-system"
+	gateway = "knative-ingressgateway"
+)
+
+// Setup creates all the initial setup needed to run the tests
+func Setup(t *testing.T) *test.Clients {
```

Comment:
Can we return error here and not accept testing as an argument? IMO its much cleaner and easier to not exit in library calls like this and perform the t.FatalF in the test method. This also causes issues when using e.g. goroutines as only one interrupt is registered for exiting.

Reply:
Can we also call this something client specific (maybe SetupClients) since this is scoped to the whole performance package.

Reply:
done